mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -436,39 +438,48 @@ fn find_pointers_and_order_post_crash(file: &mut File, max_file_size: u64) -> Ri
             read = end + data_size;
         }
 
-        // Found the last write, take whatever we got for read and write
-        // and return the pointers.
+        // Check next block
+        write_updates += 1;
+        order += 1; // We want to start with the next order number.
+        start = end + data_size;
+        end = start + block_size;
```

Comment:
Previous pr was off a bit. After getting home yesterday and running against the tests I found that the logic needed to be moved down to point to the correct block. Otherwise it points to the block before the one it should. So, moved it below the update of start.